src/ui/link/index.tsx

```diff
@@ -49,7 +68,28 @@ const Link = forwardRef<HTMLButtonElement | HTMLAnchorElement, LinkProps>(
           ''
         : to;
 
-    const handleLinkPress = (e: any) =>
+    const triggerPrefetch = useCallback(() => {
+      // ignore if async route not ready yet
+      if (typeof to !== 'string' && !route) return;
+
+      const context =
+        typeof to !== 'string' && route
+          ? createRouterContext(route, { params, query })
+          : null;
+      routerActions.prefetchNextRouteResources(linkDestination, context);
+      // omit params & query as already in linkDestination
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [route, linkDestination, routerActions]);
```

Comment:
Shouldn't `to` be in here?

Reply:
`linkDestination` covers that too, as it is a function of `to`. Will add to the comment